buildSrc/src/main/kotlin/Java9Modularity.kt

```diff
@@ -0,0 +1,63 @@
+import org.gradle.api.Project
+import org.gradle.api.tasks.bundling.Jar
+import org.gradle.api.tasks.compile.AbstractCompile
+import org.gradle.api.tasks.compile.JavaCompile
+import org.gradle.kotlin.dsl.register
+import java.util.spi.ToolProvider
+
+object Java9Modularity {
+    @JvmStatic
+    fun configureJava9ModuleInfo(project: Project) {
+        project.afterEvaluate {
+            val jdeps = ToolProvider.findFirst("jdeps").orElseThrow { IllegalStateException("Tool 'jdeps' is not available") }
```

Comment:
I'd throw this exception as late as possible (when the actual task `doLast` configured), to make sure that it won't disrupt normal local development without publishing

Reply:
Good point, I moved the `orElseThrow` to the location where `jdeps` is actually invoked.